### DISCUSSION TITLE
Minio S3 access through IAM Role

### DISCUSSION THREAD

**Discussion by Mahesh81** [minio/minio#13363] [Q&A]
Hi Team
We were able to configure minio using PVC and application able to store data . We want to configure S3 as an options, the document says
"""You can also use Amazon S3 storage instead of self-hosted MinIO's storage through passing S3 endpoint and IAM user access key ID and secret to the RP_BINARYSTORE_MINIO_* env variables, which can be defined via the same parameters in values.yaml."""

But most of the organization are restricting the IAM users/secret or access key and suggest to go with IAM Role for temporary credentials

Configuration details:
Usecase: I want to deploy minio using helm in EKS and use gateway s3 and IAM Role configured instead of accesskey and secretkey

1. I have created a EKS and deployed minio 7.1.9 using helm. 
2. Configured IAM Role in Node and Masters with AmazonS3fullaccess (Just for testing )
3.  Values.yaml has following configuration     ## Authentication configuration for S3. Ignored unless type=s3

    s3:
      accessKey: ""    ##didnt provide access and secret key as i want to connect to S3 using IAM Role attached to cluster and nodes
      secretKey: ""
      serviceEndpoint: https://s3.amazonaws.com
   
4. helm install minio --set accessKey.password=accesskey,secretKey.password=secretkey ./minio-7.1.9.tgz
5. kubectl port-forward --namespace default svc/minio 9000:9000

## Expected Behavior

1. Able to access http://127.0.0.1:9000/minio 
2. Expect to access S3 in my account from minio running in EKS

## Current Behavior

Im able to connect to node and do aws s3 ls to get list of buckets, no access issue from node 

1. Not able to view the S3 buckets from my account 
2. When i create a bucket, it uses PVC instead of s3
## Steps to Reproduce (for bugs)

.

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):

**Comment by harshavardhana** [comment]
There is literally no information here, please provide more information.  We already support IAM credentials with S3 gateway.

Moved the GitHub issue as a discussion for now.

**Reply by Mahesh81** [reply]
@harshavardhana : Updated the description

**Comment by harshavardhana** [comment]
Your helm is not something we support - our helm release is 3.1.9 I don't know where you get 7.1.9 from.
